frontend/micro-ui/web/micro-ui-internals/packages/modules/ads/src/pages/citizen/ADSMyBookings/ads-application.js

```diff
@@ -0,0 +1,94 @@
+import { Card, KeyNote, SubmitBar, Toast } from "@nudmcdgnpm/digit-ui-react-components";
+import React, { useEffect, useState } from "react";
+
+import { useTranslation } from "react-i18next";
+import { Link, useHistory } from "react-router-dom";
+
+/*
+ * AdsApplication component displays the details of a specific advertisement application.
+ * It shows key information such as booking number, applicant name, advertisement name, 
+ * booking dates, and application status. The component also includes functionality for 
+ * making payments and navigating to the application details page.
+ */
+
+const AdsApplication = ({ application, tenantId, buttonLabel }) => {
+  const { t } = useTranslation();
+  const history = useHistory();
+  const [showToast, setShowToast] = useState(null);
+//nikhil will check
+  // const { data: slotSearchData, refetch } = Digit.Hooks.chb.useChbSlotSearch({
+  //   tenantId: application?.tenantId,
+  //   filters: {
+  //     communityHallCode: application?.communityHallCode,
+  //     bookingStartDate: application?.bookingSlotDetails?.[0]?.bookingDate,
+  //     bookingEndDate: application?.bookingSlotDetails?.[application.bookingSlotDetails.length - 1]?.bookingDate,
+  //     hallCode: application?.bookingSlotDetails?.[0]?.hallCode,
+  //   },
+  //   enabled: false, // Disable automatic refetch
+  // });
+
+  const getBookingDateRange = (bookingSlotDetails) => {
+    if (!bookingSlotDetails || bookingSlotDetails.length === 0) {
+      return t("CS_NA");
+    }
+    const startDate = bookingSlotDetails[0]?.bookingDate;
+    const endDate = bookingSlotDetails[bookingSlotDetails.length - 1]?.bookingDate;
+    if (startDate === endDate) {
+      return startDate; // Return only the start date
+    } else {
+      // Format date range as needed, for example: "startDate - endDate"
+      return startDate && endDate ? `${startDate}  -  ${endDate}` : t("CS_NA");
+    }
+  };
+  const handleMakePayment = async () => {
+    const result = await refetch();
+    const isSlotBooked = result?.data?.hallSlotAvailabiltityDetails?.some((slot) => slot.slotStaus === "BOOKED");
+
+    if (isSlotBooked) {
+      setShowToast({ error: true, label: t("ADS_ADVERTISEMENT_ALREADY_BOOKED") });
+    } else {
+      history.push({
+        pathname: `/digit-ui/citizen/payment/my-bills/${"adv-services"}/${application?.bookingNo}`,
+        state: { tenantId: application?.tenantId, bookingNo: application?.bookingNo },
+      });
+    }
+  };
+  useEffect(() => {
+    if (showToast) {
```

Comment:
What are we doing here add comment

Reply:
sir in this  useEffect automatically hides the toast notification after 2 seconds by setting a timeout when showToast is active and clearing it to prevent memory leaks if the component unmounts or showToast changes, we have to do more changes in that file, me and nikhil will solves these once nikhil get this pull.